### PR TITLE
Only preserve stats for local queues

### PR DESCRIPTION
**Pull request by lukebakken**
Fixes rabbitmq/rabbitmq-management#427

Depends on rabbitmq/rabbitmq-server#1278

### CODE REVIEW

**Comment by lukebakken** [comment]
@kjnilsson - there used to be a change to [this line](https://github.com/rabbitmq/rabbitmq-management-agent/blob/rabbitmq-management-427/src/rabbit_mgmt_gc.erl#L86) to use `rabbit_amqqueue:list_local_names()`. Is this no longer needed?

**Comment by kjnilsson** [comment]
@lukebakken that is correct. Management GC needs to use the cluster-wide list of queues in order to retain historical queue metrics for queues that used to reside on the current node.

File: src/rabbit_mgmt_metrics_collector.erl [hunk 1]
```diff
@@ -140,19 +140,29 @@ take_smaller(Policies) ->
...
                                           NegStats, Size, Interval, true)
-                             || {Size, Interval} <- GPolicies]
+                             || {Size, Interval} <- GPolicies],
+                            % zero out msg stats to avoid duplicating msg
+                            % stats when a master queue is migrated
+                            QNegStats = ?queue_msg_stats(0, 0, 0),
+                            [insert_entry(queue_msg_stats, Queue, TS,
+                                          QNegStats, Size, Interval, false)
+                             || {Size, Interval} <- BPolicies],
+                            % these need to be delayed by an interval
```
Comment (lukebakken) [inline src/rabbit_mgmt_metrics_collector.erl]: @kjnilsson - is this comment still relevant? i.e, should a delay be introduced?

Comment (kjnilsson) [inline src/rabbit_mgmt_metrics_collector.erl]: Good point that can be removed.

**Pull request merged**